File: src/main/java/com/github/alexandrecarlton/idea/settings/layout/project_settings/modules/ModuleSettings.java
package com.github.alexandrecarlton.idea.settings.layout.project_settings.modules;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;

import org.immutables.value.Value;

import java.util.List;

@Value.Immutable
@JsonDeserialize(as = ImmutableModuleSettings.class)
public interface ModuleSettings {

  String name();

  List<ModuleSourceSettings> sources();

}

File: src/main/kotlin/com/github/alexandrecarlton/idea/settings/layout/project_settings/modules/ModuleSettings.kt
package com.github.alexandrecarlton.idea.settings.layout.project_settings.modules

import com.fasterxml.jackson.annotation.JsonProperty

data class ModuleSettings(

    @JsonProperty("Name")
    val name: String,

    @JsonProperty("Sources")
    val sources: List<ModuleSourceSettings>? = null
)
